### PR TITLE
Feature general fixes

### PR DESCRIPTION
**Pull request by tegefaulkes**
### Description

This PR is just a collection of general small fixes in `Polykey`.

### Final checklist

* [x] Domain specific tests
* [x] Full tests
* [x] Updated inline-comment documentation
* [x] Lint fixed
* [x] Squash and rebased
* [x] Sanity check the final build

### CODE REVIEW

**Comment by CMCDragonkai** [comment]
Usually random fixes like this I'd just push to staging. But shouldn't this all be part of issues?

**Comment by tegefaulkes** [comment]
I'd agree with that but I wanted a paper trail for what I was doing. I don't think there are existing issues for any of theses. 

The fix for the intermittent test failure has an existing issue related to it. But this far from fully addresses that.

File: src/nodes/NodeConnectionManager.ts [hunk 1]
```diff
@@ -1278,9 +1287,15 @@ class NodeConnectionManager {
...
       });
-    }).finally(() => {
-      this.activeSignalFinalPs.delete(connProm);
-    });
+    })
+      // Ignore results and failures, then are expected to happen and are allowed
+      .then(
+        () => {},
+        () => {},
+      )
```
Comment (tegefaulkes) [inline src/nodes/NodeConnectionManager.ts]: I should really only ignore connection specific failures here. I'm going to fix that quickly.

Comment (tegefaulkes) [reply src/nodes/NodeConnectionManager.ts]: Addressed, I'm having it check if the error is connection related and throwing if not.

**Pull request merged**